java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -450,7 +498,7 @@ public HostColumnVector copyToHost() {
       BaseDeviceMemoryBuffer valid = getValidityBuffer();
       BaseDeviceMemoryBuffer offsets = getOffsetBuffer();
       BaseDeviceMemoryBuffer data = null;
-      DType type = this.type;
+      DType type = this.type.typeId;
```

Comment:
Shouldn't this be using `this.type` here?  Once the `isNestedType` predicate is added, we will need to pass a `DataType` to the `HostColumnVector` constructor so it can access the scale to properly decode decimal types.